README.asciidoc

```diff
@@ -0,0 +1,398 @@
+* jit_functions - used it in image processing, jit makes for loops much faster so we can go through lists much faster
+* numpy - for list manipulation
+* cv2 - image processing
+* tkinter - used it in the beginning to simulate omnimotion visually
+* websockets - to make our testing server for referee commands
+
+=== A brief description of your game logic and overall code structure.
+
+Our code overall is section into files according to their usage (for example immage_processing.py and movement.py) to make the code less clutterd then if it was all in the same file. The fails contain mainly different classes and then we can import them according to our needs. The main code is structured so that every situatsion has a State that we can refere to to make the robot move accordingly.
+
+=== Block diagram of game logic
+image::https://user-images.githubusercontent.com/112713620/212192438-b3be92dc-7776-4a3d-9623-ac57508f8613.JPG[750,750]
+
+=== Good and bad
+
+* One of our code's strong side is sigmoid controller. It eliminated most of the if else sentences and also made the robot's movement better.
+* Ball dedection and depth dedection could be improved by using better and more calculations.
+* Timer that we used could be made into a function to make the code more compact.
+* Line dedection is very reliable.
+* Code was good enough to win DeltaX.
+
+
+== Electronics
+
+The electronics of the robot use the components that were recommended and provided by the instrucors.
+The mainboard can control 3 motors and 2 servos. The motor drivers are powered by a 4S LiPo battery and the microcontroller side is powered through the usb connector by the NUC computer. The USB connection is also used to communicate between the microcontroller and the computer. There is a connector for an IR sensor to detect the ball in the holder. The microcontroller will let the computer know when the IR sensor detects a ball.
+
+=== Some problems that occured and the final outcome
+* After soldering all the components, there was a problem with the 5V on the battery side of the board. It turned out that one of the pins of the switching regulator was not connected.
+* Some of the PWM and DIR pins connections were swapped. Needed to hack the board to fix it.
+* Micro USB connector broke and had to be replaced.
+* IR sensor wire broke during the robot assembly and it needed to be fixed.
+
+* Everyting worked in the end and we won Delta-X and the course competition.
+
+image::https://user-images.githubusercontent.com/112662827/211169992-68409b84-9493-466a-b800-7c67e5b26b65.jpg[500,500]
+image::https://user-images.githubusercontent.com/112662827/211169998-20ac7f89-c637-45fc-b4ae-4e3fb1bdacac.jpg[500,500]
+image::https://user-images.githubusercontent.com/112662827/211170003-e3264e73-28b2-42e0-b00a-e51ef2a45c92.jpg[500,500]
+image::https://user-images.githubusercontent.com/112662827/211170010-b454fe8b-06f9-461d-827f-3709ff7763a9.jpg[950,500]
+
+== Mechanics

```

Comment:
Missing:
* A short analysis of what was good and bad in your mechanics and how to improve.